RecoJets/JetProducers/plugins/HFJetShowerShape.cc

```diff
@@ -0,0 +1,229 @@
+      int size_CentralEtaStrip(0.), size_AdjacentEtaStrips(0.);
+      double sigmaEtaEtaSq(0.), sigmaPhiPhiSq(0.);
+      double sumweightsPFcands = 0;
+      for (unsigned i = 0; i < jet.numberOfSourceCandidatePtrs(); ++i) {
+        const reco::Candidate* icand = jet.sourceCandidatePtr(i).get();
+        //Only look at pdgId =1,2 (HF PF cands)
+        if (std::abs(icand->pdgId()) > 2)
+          continue;
+
+        double deta = std::abs(icand->eta() - jet.eta());
+        double dphi = std::abs(deltaPhi(icand->phi(), jet.phi()));
+        double pt_PUsub = icand->pt() - puoffset;
+
+        //This is simply the size of the central eta strip and the adjacent strips
+        if (pt_PUsub >= stripPtThreshold_) {
+          if (dphi <= hfTowerPhiWidth_ * 0.5)
+            size_CentralEtaStrip++;
+          else if (dphi <= hfTowerPhiWidth_ * 1.5)
+            size_AdjacentEtaStrips++;
+        }
+
+        //Now computing sigmaEtaEta/PhiPhi
+        if (pt_PUsub >= widthPtThreshold_ && sumptPFcands > 0) {
+          double weight = pt_PUsub / sumptPFcands;
+          sigmaEtaEtaSq += deta * deta * weight;
+          sigmaPhiPhiSq += dphi * dphi * weight;
+          sumweightsPFcands += weight;
+        }
+      }
+
+      v_size_CentralEtaStrip.push_back(size_CentralEtaStrip);
+      v_size_AdjacentEtaStrips.push_back(size_AdjacentEtaStrips);
+
+      if (sumweightsPFcands > 0 && sigmaEtaEtaSq > 0 && sigmaPhiPhiSq > 0) {
+        v_sigmaEtaEta.push_back(sqrt(sigmaEtaEtaSq / sumweightsPFcands));
+        v_sigmaPhiPhi.push_back(sqrt(sigmaPhiPhiSq / sumweightsPFcands));
+      } else {
+        v_sigmaEtaEta.push_back(-1.);
+        v_sigmaPhiPhi.push_back(-1.);
+      }
+
+    }  //End loop over jets
+  }
+
+  putInEvent("sigmaEtaEta", theJets, v_sigmaEtaEta, iEvent);
+  putInEvent("sigmaPhiPhi", theJets, v_sigmaPhiPhi, iEvent);
+  putInEvent("centralEtaStripSize", theJets, v_size_CentralEtaStrip, iEvent);
+  putInEvent("adjacentEtaStripsSize", theJets, v_size_AdjacentEtaStrips, iEvent);
+}
+
+/// Function to put product into event
+template <typename T>
+void HFJetShowerShape::putInEvent(const std::string& name,
+                                  const edm::Handle<edm::View<reco::Jet>>& jets,
+                                  std::vector<T> product,
+                                  edm::Event& iEvent) const {
+  auto out = std::make_unique<edm::ValueMap<T>>();
+  typename edm::ValueMap<T>::Filler filler(*out);
+  filler.insert(jets, product.begin(), product.end());
+  filler.fill();
+  iEvent.put(std::move(out), name);
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void HFJetShowerShape::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("theJets", edm::InputTag("ak4PFJetsCHS"));
+  desc.add<edm::InputTag>("theVertices", edm::InputTag("offlinePrimaryVerticesWithBS"));
```

Comment:
BTW, is  the "the" prefix all that necessary, wouldn't `jets` and `vertices` be already clear enough?